base/include/distributed/comms_mpi_hostbuffer_stream.h

```diff
@@ -501,7 +501,8 @@ class CommsMPIHostBufferStream : public CommsMPI<T_Config>
         void get_hostname(std::string &my_hostname);
         void exchange_hostnames(std::string &my_hostname, std::vector<std::string> &hostnames, int num_parts );
 
-        void all_gather(IndexType_h &my_data, HIVector &gathered_data, int num_parts);
+        void all_gather(const IndexType_h &my_data, HIVector &gathered_data, int num_parts);
```

Comment:
Wouldn't those two methods interfere if one would instantiate AMGX with IndexType = int64_t ? (related https://github.com/NVIDIA/AMGX/issues/145)

Reply:
Yes I think you might be right there - I'll adjust.

Reply:
Hopefully this issue is fixed with the most recent commit.